src/wiringX.c

```diff
@@ -425,6 +428,181 @@ int wiringXSPISetup(int channel, int speed) {
 	return -1;
 }
 
+int wiringXSerialOpen(char *device, struct wiringXSerial_t wiringXSerial) {
+	struct termios options;
+	speed_t myBaud;
+	int status, fd;
+
+	switch(wiringXSerial.baud) {
+		case     50:	myBaud = B50; break;
+		case     75:	myBaud = B75; break;
+		case    110:	myBaud = B110; break;
+		case    134:	myBaud = B134; break;
+		case    150:	myBaud = B150; break;
+		case    200:	myBaud = B200; break;
+		case    300:	myBaud = B300; break;
+		case    600:	myBaud = B600; break;
+		case   1200:	myBaud = B1200; break;
+		case   1800:	myBaud = B1800; break;
+		case   2400:	myBaud = B2400; break;
+		case   4800:	myBaud = B4800; break;
+		case   9600:	myBaud = B9600; break;
+		case  19200:	myBaud = B19200; break;
+		case  38400:	myBaud = B38400; break;
+		case  57600:	myBaud = B57600; break;
+		case 115200:	myBaud = B115200; break;
+		case 230400:	myBaud = B230400; break;
+
+		default:
+		return -2;
+	}
+
+	if((fd = open(device, O_RDWR | O_NOCTTY | O_NDELAY | O_NONBLOCK)) == -1) {
```

Comment:
I was about to merge the serial interface when i noticed you added analogRead. Are you going to add analogWrite also?